projects/plugins/boost/app/modules/cache/pre-wordpress/Boost_Cache.php

```diff
@@ -193,13 +193,25 @@ public function delete_on_comment_edit( $comment_id, $commentdata ) {
 	 */
 	public function delete_on_comment_post( $comment_id, $comment_approved, $commentdata ) {
 		$post = get_post( $commentdata['comment_post_ID'] );
-
+		error_Log( "delete_on_comment_post: $comment_id, $comment_approved, {$post->ID}" ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
 		/**
 		 * If a comment is not approved, we only need to delete the cache for
 		 * this post for this visitor so the unmoderated comment is shown to them.
 		 */
 		if ( $comment_approved !== 1 ) {
-			$this->storage->invalidate_single_visitor( Boost_Cache_Utils::normalize_request_uri( get_permalink( $post->ID ) ), $this->request->get_parameters() );
+			$parameters = $this->request->get_parameters();
+			/**
+			 * if there are no cookies, then visitor did not click "remember me".
+			 * No need to delete the cache for this visitor as they'll be
+			 * redirected to a page with a hash in the URL for the moderation
+			 * message.
+			 */
+			if ( isset( $parameters['cookies'] ) && ! empty( $parameters['cookies'] ) ) {
+				$this->storage->invalidate(
+					Boost_Cache_Utils::normalize_request_uri( get_permalink( $post->ID ) ),
+					'/**/' . Boost_Cache_Utils::get_request_filename( $this->request->get_parameters() )
```

Comment:
I would extract `$hash` variable from this and do `"/**/{$hash}"` as the pattern

Reply:
Done, looks better.